### ISSUE TITLE
[CPU shaders] Assertion failure in path_tiling stage when rendering the mmark test scene

### ISSUE THREAD

**Issue by armansito** [linebender/vello#379]
To reproduce, run the following and press the right arrow key to navigate to the motion mark scene:
```
cargo run -p with_winit -- --test-scenes  --use-cpu
```
This fails with: `thread 'main' panicked at 'assertion failed: xy0.x >= tile_xy.x && xy0.x <= tile_xy1.x', src/cpu_shader/path_tiling.rs:129:9`

With some more debug information:
```
thread 'main' panicked at 'p0: Vec2 {
    x: NaN,
    y: NaN,
}, p1: Vec2 {
    x: NaN,
    y: NaN,
}, xy0: Vec2 {
    x: NaN,
    y: NaN,
}, xy1: Vec2 {
    x: NaN,
    y: NaN,
}, tile_xy: Vec2 {
    x: 0.0,
    y: 0.0,
}, tile_xy1: Vec2 {
    x: 16.0,
    y: 16.0,
}', src/cpu_shader/path_tiling.rs:129:9
```

**Comment by raphlinus** [comment]
This should be fixed by https://github.com/linebender/kurbo/pull/314

**Comment by DJMcNab** [comment]
I believe this should have been fixed by #396 - I certainly cannot reproduce on my system

**Comment by raphlinus** [comment]
Yes, closed by #396. Certainly it's possible that there are more bugs in strokes, but those should be filed as separate issues (ideally under kurbo, where ideally we'll put some property testing in place as well).
